elfi/methods.py

```diff
@@ -266,18 +265,23 @@ def reject(self, threshold, n_sim=None):
 
         distances, parameters = self._acquire(n_sim)
         accepted = distances[:, 0] < threshold
-        samples = [p[accepted] for p in parameters]
-        distances = distances[accepted]
         accept_rate = sum(accepted)/n_sim
 
-        return dict(samples=samples,
-                    distances=distances,
-                    threshold=threshold,
-                    n_sim=n_sim,
-                    accept_rate=accept_rate)
+        samples = OrderedDict()
+        for ii, p in enumerate(parameters):
+            samples[self.parameter_nodes[ii].name] = p[accepted]
+        distances = distances[accepted]
+
+        result = Result(samples=samples,
+                        distances=distances,
+                        threshold=threshold,
+                        n_sim=n_sim,
+                        accept_rate=accept_rate)
+
+        return result
 
 
-# TODO: add tests
+# TODO: add tests, docs
```

Comment:
True in general, but this particular TODO belongs to an already merged function..